### PR TITLE
Dedup CupertinoAlertDialog and CupertinoActionSheet source code

### LINKED ISSUE #20470: Dedup similar code between CupertinoAlertDialog and CupertinoActionSheet
After #18381 is merged, dedup similar code between CupertinoAlertDialog and CupertinoActionSheet.

The two were written in parallel by different developers. The Action Sheet copied a bunch of code from the Alert Dialog work and that copied code needs to be dedup'd.

### PR DESCRIPTION
**Pull request by campovski**
Files `packages/flutter/lib/src/cupertino/action_sheet.dart` and `packages/flutter/lib/src/cupertino/dialog.dart` shared a lot of similarities. The code was deduped and `CupertinoActionSheet` was moved to `packages/flutter/lib/src/cupertino/dialog.dart` from `packages/flutter/lib/src/cupertino/action_sheet.dart` (tests were moved in similar fashion as well, grouping previous tests from `dialog_test.dart` into `group('CupertinoAlertDialog', () {...});` and those from `action_sheet_test.dart` into `group('CupertinoActionSheet', () {...});`. No new tests were introduced because previous tests already cover the affected classes sufficiently.

Deduping was done by merging of the private classes in both files, pushing differences (like stlyes and paddings) to class fields and then injecting them from the parent classes depending on whether dialog or action sheet is built. Similarly, where necessary, an auxiliary field `isActionSheet` was introduced to denote whether a dialog or action sheet has invoked the construction and is used in places where methods slightly defer for each of the aforementioned widgets, thus making it unnecessary to separate the code into two different classes.

compare

Fixes #20470 

`flutter/tests` repo left intacted.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by campovski** [comment]
@Hixie I assume this change is test-exempt because existing tests seem sufficient. Please confirm or deny.

**Comment by Hixie** [comment]
It's probably test-exempt but it's hard to tell because of the formatting differences. Would it be possible to reconstruct the PR as two commits, one being just literally the code move with zero code changes (which wouldn't compile) and one being the edits required to make this work, without any `dartfmt` changes? That would make reviewing the PR much easier.

See also https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting

**Comment by campovski** [comment]
Yes, of course. And my appologies for using `dartfmt`, I must have forgotten about the formatting paragraph since this is my first contribution and there is quite a lot of documentation to remember everything. I will learn everything along the way 😊

**Comment by campovski** [comment]
@Hixie Done. I reformatted the PR into two commits. First one only copies the code from `action_sheet(_test).dart` to `dialog(_test).dart` and performs minimal changes like renaming of constants, and second one dedupes the code. I also disabled `dartfmt` and the diff is way more readable now.

File: packages/flutter/test/cupertino/dialog_test.dart
Comment (Hixie) [inline packages/flutter/test/cupertino/dialog_test.dart:12]: It's fine to keep the tests in two separate files. In fact that's probably better -- ideally this PR wouldn't touch the tests, that's the best way to show that it didn't break anything.

File: packages/flutter/lib/src/cupertino/dialog.dart
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/dialog.dart:160]: this change is surprising. Is this commit exclusively moving code around?

File: packages/flutter/lib/src/cupertino/dialog.dart
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/dialog.dart]: similarly with this change, I'm surprised to see this in a commit that's just moving code

**Comment by Hixie** [comment]
Thanks for splitting the commits, that's definitely helpful. Sorry for being a pain, but in reviewing the PR it seems like there's still more going on in the first commit than I would expect for just a code move. It looks like maybe there should be more commits to make it clearer? e.g. maybe one commit to prepare the code (e.g. renaming constants), then one commit that is literally just a cut-and-paste, with no code changes at all, then a commit to do the deduping?

**Comment by campovski** [comment]
Yes, I left some minor changes and documentation improvements in the first commit. Sorry for not making self-contained commits, will split into at least three commits.

Regarding leaving the tests in the same file: I can leave them as they are, I just do not like the idea of having `action_sheet_test.dart` file when there is no `action_sheet.dart`. This is just my practice; that is why I moved the tests.

**Comment by Hixie** [comment]
> Regarding leaving the tests in the same file: I can leave them as they are, I just do not like the idea of having `action_sheet_test.dart` file when there is no `action_sheet.dart`. This is just my practice; that is why I moved the tests.

Yeah definitely don't worry about that. There's no particular mapping of test file to source code. Lots of things that we have to test cover multiple files, and individual files have lots of test files that cover them.

> Yes, I left some minor changes and documentation improvements in the first commit. Sorry for not making self-contained commits, will split into at least three commits.

Thanks, and sorry again for making you jump through hoops. It's much appreciated.

**Comment by campovski** [comment]
> Yeah definitely don't worry about that. There's no particular mapping of test file to source code. Lots of things that we have to test cover multiple files, and individual files have lots of test files that cover them.

Yeah, I saw it when I was working on `flutter_tool`. Left the tests as they are.

> Thanks, and sorry again for making you jump through hoops. It's much appreciated.

No worries, glad I could help.

**Comment by Hixie** [comment]
Thanks so much for organizing the code to be easy to review. I'll leave the actual review to someone on @HansMuller's team since they will be more familiar with this code.

**Comment by Hixie** [comment]
Oh sorry forgot to give a call on test-exemptness! Yes, I believe this is probably test-exempt since it's "just" a refactoring (a valuable one). The reviewer may call out some things they'd like tested explicitly though, in case the existing tests are insufficient.

**Comment by campovski** [comment]
Thanks, will leave the checkbox unchecked for now and once I get LGTM I will check it.

File: packages/flutter/lib/src/cupertino/dialog.dart
Comment (justinmc) [inline packages/flutter/lib/src/cupertino/dialog.dart:1138]: The formatting should stay as it was originally, the indentation should be 2.  See [here in the style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#indent-multi-line-argument-and-parameter-lists-by-2-characters) (there is an example of a List just below that section).

Does your IDE have a formatter that ran on this code maybe?  I'll try to keep an eye out for other formatting changes as I review this.

File: packages/flutter/lib/src/cupertino/dialog.dart
Comment (justinmc) [inline packages/flutter/lib/src/cupertino/dialog.dart:1341]: Here's another formatting change.  The space between `{` and `required` should be there.  I can't find this in the style guide explicitly but it's clear in many examples.

File: packages/flutter/lib/src/cupertino/dialog.dart
Comment (justinmc) [inline packages/flutter/lib/src/cupertino/dialog.dart:1380]: I see there was an indentation change here, too.  I can't tell from this diff if the closing parenthesis here is aligned with `result.addWithPaintOffset`, but that's how it should be.  If it is, then this change is fine.

File: packages/flutter/lib/src/cupertino/dialog.dart
Comment (justinmc) [inline packages/flutter/lib/src/cupertino/dialog.dart:1812]: Extra indentation like this is fine either way I think.

**Review by justinmc — COMMENTED** [review]
There are a few formatting changes that probably shouldn't have been made.  I think you may be running an autoformatter that isn't the same as the Flutter style guide.  I commented on a few but probably missed some.

Otherwise this looks good!  Thanks for cleaning this up.

File: packages/flutter/lib/src/cupertino/dialog.dart
Comment (campovski) [reply packages/flutter/lib/src/cupertino/dialog.dart:1138]: No, I disabled all formatting, probably this was done by my mistake.

File: packages/flutter/lib/src/cupertino/dialog.dart
Comment (campovski) [reply packages/flutter/lib/src/cupertino/dialog.dart:1812]: Yes, I saw in cases like this both 2 and 4 space indents are used. Since I prefer 4 I used it.

**Review by justinmc — APPROVED** [review]
LGTM 👍   Thanks for the fixes.  I'll try to run the Google tests on this before merging.

**Comment by justinmc** [comment]
There was a failing Google test but it seems to have been a flake.  I've added the label to automerge this when possible.

**Comment by campovski** [comment]
@justinmc The PR seems ready to be merged now.

**Comment by campovski** [comment]
@justinmc This can be merged now I think. 😊

**Comment by justinmc** [comment]
Google tests have passed 👍

**Pull request merged**
